pkg/controller/master/virtualmachine/vmi_controller.go

```diff
@@ -109,3 +110,38 @@ func (h *VMIController) removeDeprecatedFinalizer(_ string, vmi *kubevirtv1.Virt
 	}
 	return vmi, nil
 }
+
+// ReconcileFromHostLabels handles the propagation of metadata from node labels to VirtualMachineInstance annotations.
+func (h *VMIController) PatchMacAddresses(_ string, vmi *kubevirtv1.VirtualMachineInstance) (*kubevirtv1.VirtualMachineInstance, error) {
+	if vmi == nil || (vmi.DeletionTimestamp == nil && !vmi.IsFinal()) {
+		return vmi, nil
+	}
+
+	vm, err := h.virtualMachineCache.Get(vmi.Namespace, vmi.Name)
+	if err != nil {
+		return vmi, err
+	}
+
+	macAddressStr := vm.Annotations[util.AnnotationMacAddressName]
+	if macAddressStr == "" {
+		return vmi, nil
+	}
+
+	macAddresses := map[string]string{}
+	if err = json.Unmarshal([]byte(macAddressStr), &macAddresses); err != nil {
```

Comment:
For this error, maybe a better way is to log an error message, and then return (like there is no MAC annotation), otherwise, the VM will be blocked, reconciller won't solve such case